### ISSUE TITLE
[package] automake/1.16.2 and libtool/2.4.6: no Linux binary 

### ISSUE THREAD

**Issue by ericLemanissier** [conan-io/conan-center-index#2196]
When trying to consume automake or libtool, automake and libtool binaries are always missing, even if https://conan.io/center/automake/1.16.2/?user=_&channel=_&revision=&os=&tab=config has a linux configuration and https://conan.io/center/libtool/2.4.6/?user=_&channel=_&revision=&os=Linux&tab=config too
Maybe the binaries of these packages need to be regenerated after [m4 was added to autoconf's requirement](https://github.com/conan-io/conan-center-index/pull/2103/files) ?

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **automake/1.16.2**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **GCC 7**
  * Conan version: **conan 1.27.1**
  * Python version: **Python 3.6.9**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
`conan install automake/1.16.2@`

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Installing package: automake/1.16.2
Requirements
    autoconf/2.69 from 'conan-center' - Cache
    automake/1.16.2 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
Packages
    autoconf/2.69:981def0a8607e5b81282ae8da6b772973e6c0636 - Download
    automake/1.16.2:8f0570538ddd703bfc5400385a2fca9aa4a2b671 - Missing
    m4/1.4.18:efbe354690ef83824a1e98c7a6076b7ab63bc1b3 - Cache

Installing (downloading, building) binaries...
ERROR: Missing binary: automake/1.16.2:8f0570538ddd703bfc5400385a2fca9aa4a2b671

automake/1.16.2: WARN: Can't find a 'automake/1.16.2' package for the specified settings, options and dependencies:
- Settings: os=Linux
- Options:
- Dependencies: autoconf/2.69
- Requirements: autoconf/2.Y.Z
- Package ID: 8f0570538ddd703bfc5400385a2fca9aa4a2b671

ERROR: Missing prebuilt package for 'automake/1.16.2'
Try to build from sources with "--build=automake"
Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
```

</details>

### Steps to reproduce (Include if Applicable)
`conan install libtool/2.4.6@`

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```

Installing package: libtool/2.4.6
Requirements
    autoconf/2.69 from 'conan-center' - Cache
    automake/1.16.2 from 'conan-center' - Cache
    libtool/2.4.6 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
Packages
    autoconf/2.69:981def0a8607e5b81282ae8da6b772973e6c0636 - Download
    automake/1.16.2:8f0570538ddd703bfc5400385a2fca9aa4a2b671 - Missing
    libtool/2.4.6:3726a3c7a7e9851261a861de35a0f9639204144e - Missing
    m4/1.4.18:efbe354690ef83824a1e98c7a6076b7ab63bc1b3 - Cache

Installing (downloading, building) binaries...
ERROR: Missing binary: automake/1.16.2:8f0570538ddd703bfc5400385a2fca9aa4a2b671
ERROR: Missing binary: libtool/2.4.6:3726a3c7a7e9851261a861de35a0f9639204144e

automake/1.16.2: WARN: Can't find a 'automake/1.16.2' package for the specified settings, options and dependencies:
- Settings: os=Linux
- Options:
- Dependencies: autoconf/2.69
- Requirements: autoconf/2.Y.Z
- Package ID: 8f0570538ddd703bfc5400385a2fca9aa4a2b671

ERROR: Missing prebuilt package for 'automake/1.16.2', 'libtool/2.4.6'
Try to build from sources with "--build=automake --build=libtool"
Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
```

</details>

**Comment by uilianries** [comment]
@jgsogo @danimtb Indeed the package is missing, I can reproduce the same behavior.

automake recipe ONLY uses settings.os for its package ID, no options. However, it contains `autoconf/2.69` as requirement.

The last change was 10 days ago (https://github.com/conan-io/conan-center-index/pull/2038), but only generated 3 packages:

Linux:   258f6f05ca54813b3d7180161753df7f9ccf7e1d
Macos:   3c28ba69276b46fd46e9c2060ba991d725ccab43
Windows: 3e48e69237f7f2196164383ef9dedf0f93cbf249

The current Conan search outputs the same:

```
conan search -r conan-center "automake/1.16.2@"
Existing packages for recipe automake/1.16.2:

Existing recipe in remote 'conan-center':

    Package_ID: 258f6f05ca54813b3d7180161753df7f9ccf7e1d
        [settings]
            os: Linux
        [requires]
            autoconf/2.69:44fcf6b9a7fb86b2586303e3db40189d3b511830
        Outdated from recipe: False

    Package_ID: 3c28ba69276b46fd46e9c2060ba991d725ccab43
        [settings]
            os: Macos
        [requires]
            autoconf/2.69:743cf0321be3152777da4d05247a66d1552e70a2
        Outdated from recipe: False

    Package_ID: 3e48e69237f7f2196164383ef9dedf0f93cbf249
        [settings]
            os: Windows
        [requires]
            autoconf/2.69:456f15897172eef340fcbac8a70811f2beb26a93
        Outdated from recipe: False
```

The only recent change that I found was a new transitive dependency (https://github.com/conan-io/conan-center-index/pull/2103).

The behavior occurs with Conan 1.25, 1.26 and 1.27. So, for some reason, the CCI generated a different package ID that Conan client requested for Linux.

**Comment by ericLemanissier** [comment]
@uilianries [with this change ](https://github.com/conan-io/conan-center-index/pull/2197/files#diff-ce685a87b04ba191af305140cead2646R79-R82) the binary hashes get back to what they were before, so this PR should solve the issue right?

**Comment by uilianries** [comment]
@ericLemanissier  Indeed! I didn't read that PR yet, thanks for linking it.

**Comment by jgsogo** [comment]
Yes, adding a requirement modifies the package-ID which turns out in missing packages in the graph. Once again, we need a job running every hour/day checking for these missing packages and triggering a build once detected. It is in the backlog (codename "tapaholes").

I suggested a different approach for the PR linked.

Thanks for noticing and opening the issue 🙌

**Comment by danimtb** [comment]
@jgsogo for future reference, we have to verify this workaround is fine or should be revisited https://github.com/conan-io/conan-center-index/pull/2197

**Comment by danimtb** [comment]
This should be "fixed" or workarounded by https://github.com/conan-io/conan-center-index/pull/2197

Other PRs affected: https://github.com/conan-io/conan-center-index/pull/2167#issuecomment-660910557, https://github.com/conan-io/conan-center-index/pull/2220#issuecomment-661240511

**Comment by madebr** [comment]
@ericLemanissier 
I think this is fixed and can be closed.

**Comment by madebr** [comment]
@danimtb 

> Other PRs affected: [#2167 (comment)](https://github.com/conan-io/conan-center-index/pull/2167#issuecomment-660910557), [#2220 (comment)](https://github.com/conan-io/conan-center-index/pull/2220#issuecomment-661240511)


Thos pr's were/are plagued by another problem that is solved by #2314
